Capture debug-image if present

keithc-ca opened the pull request:
This is meant to address #1596.

Also captures test-image for jdk8.

karianna commented on configureBuild.sh:
```diff
@@ -119,7 +119,8 @@ setVariablesForConfigure() {
   # test-image target is JDK 11+ only. Set to the empty string
   # as build scripts check whether the variable is non-empty.
-  local openjdk_test_image_path=""
+  local openjdk_test_image_path="test"
```
I think this change from the non-empty string will cause issues in later scripts (see the comment above).

keithc-ca replied:
As I understand this, the only difference here is to enable the test-image for jdk8 like other versions. Likewise for docker-build.sh.

karianna replied:
OK I updated the comment inline

johnoliver replied:
in the java 8 repo there is a dir called test:

https://github.com/AdoptOpenJDK/openjdk-jdk8u/tree/master/test

I think this change will have the effect of packaging and archiving that dir

karianna commented on docker-build.sh:
```diff
@@ -83,7 +83,8 @@ buildOpenJDKViaDocker()
 
     local openjdk_core_version=${BUILD_CONFIG[OPENJDK_CORE_VERSION]}
-    local openjdk_test_image_path=""
+    local openjdk_test_image_path="test"
```
Same comment as above.

keithc-ca commented:
> One question and does the README also need updating?

I didn't see an obvious place to add information there about this change.

sxa555 commented:
Have queued up https://ci.adoptopenjdk.net/view/work%20in%20progress/job/VagrantPlaybookCheck/434/ to verify if it is likely to break anything in the pipelines (FYI @Willsparker )

gdams commented:
run tests

keithc-ca commented:
> On reflection, default of test will break Java 8 builds.

I'm not sure why this claim is being made. In any case, perhaps I should revert the test-related changes so we can focus on the debug image for now?

karianna commented:
@johnoliver can explain better but I think that because the Java 8 builds have a pre-existing test folder there's a clash...

johnoliver commented:
@keithc-ca that value is written into `BUILD_CONFIG[DEBUG_IMAGE_PATH]`
```
BUILD_CONFIG[DEBUG_IMAGE_PATH]=$openjdk_debug_image_path
```
Now that DEBUG_IMAGE_PATH is set the following will now be true when it was previously not:

https://github.com/AdoptOpenJDK/openjdk-build/blob/298421668973dc66c14cf220dbc04fea03f0d1c9/sbin/build.sh#L533

Is that what you intended to do with that change, and is that behaviour is correct? If so, should be fine but the previous comments on this indicate that was only meant to be for java 11+

karianna commented:
> @keithc-ca that value is written into `BUILD_CONFIG[DEBUG_IMAGE_PATH]`
> 
> ```
> BUILD_CONFIG[DEBUG_IMAGE_PATH]=$openjdk_debug_image_path
> ```
> 
> Now that DEBUG_IMAGE_PATH is set the following will now be true when it was previously not:
> 
> https://github.com/AdoptOpenJDK/openjdk-build/blob/298421668973dc66c14cf220dbc04fea03f0d1c9/sbin/build.sh#L533
> 
> Is that what you intended to do with that change, and is that behaviour is correct? If so, should be fine but the previous comments on this indicate that the logic of that if statement was only meant to be for java 11+ only

@johnoliver did you mean TEST_IMAGE_PATH in your above comment?

johnoliver commented:
@karianna yes I did

keithc-ca commented:
Both `${BUILD_CONFIG[TEST_IMAGE_PATH]}` and `${BUILD_CONFIG[DEBUG_IMAGE_PATH]}` are relative paths and will be interpreted based on the current directory set at [1] (`build/*/images`).
All jdk versions may have `images/test`, so I'm still not sure why this would be a 'clash'.
The primary goal of this PR is to also recognize `images/debug-image`.

[1] https://github.com/AdoptOpenJDK/openjdk-build/blob/master/sbin/build.sh#L509

keithc-ca commented:
Updated and rebased.
* updated comments
* use local variables for better clarity
* don't use `$(ls -d $dir)` when `$dir` will not contain wildcards; just use `$dir`

johnoliver commented:
looks good
